mp/src/vgui2/vgui_controls/CvarComboBox.cpp

```diff
@@ -0,0 +1,158 @@
+//========= Copyright Valve Corporation, All rights reserved. ============//
+//
+// Purpose: 
+//
+// $NoKeywords: $
+//=============================================================================//
+
+#include "tier1/KeyValues.h"
+
+#include <vgui_controls/CvarComboBox.h>
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include <tier0/memdbgon.h>
+
+using namespace vgui;
+
+Panel *Create_CvarComboBox()
+{
+    return new CvarComboBox( nullptr, nullptr );
+}
+
+//not sure which build factory to use here (and if the constructor above is necessary)
+DECLARE_BUILD_FACTORY_CUSTOM( CvarComboBox, Create_CvarComboBox );
+
+//-----------------------------------------------------------------------------
+// Purpose: Constructor
+//-----------------------------------------------------------------------------
+CvarComboBox::CvarComboBox(Panel *parent, const char *panelName, int numLines, bool allowEdit, char const *cvarname, bool ignoreMissingCvar)
+    : ComboBox(parent, panelName, numLines, allowEdit), m_cvar((cvarname) ? cvarname : "", (cvarname) ? ignoreMissingCvar : true)
+{
+    InitSettings();
+    m_bIgnoreMissingCvar = ignoreMissingCvar;
+
+    if (m_cvar.IsValid())
+    {
+        Reset();
+    }
+    AddActionSignalTarget(this);
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Destructor
+//-----------------------------------------------------------------------------
+CvarComboBox::~CvarComboBox()
+{
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: 
+//-----------------------------------------------------------------------------
+void CvarComboBox::Paint()
+{
+    if (!m_cvar.IsValid())
+    {
+        BaseClass::Paint();
+        return;
+    }
+
+    int value = m_cvar.GetInt();
+
+    if (value != m_iStartValue)
+    {
+        ActivateItemByRow(value);
+        m_iStartValue = value;
+    }
```

Comment:
Similarly, this should be moved to `OnThink()` since it's not really doing anything painting related, but rather setting internal logic state every frame.

That will effectively get rid of this 'Paint` method as it'll only call the base class function regardless of if the cvar is valid or not.

Make sure to check the validity of the cvar in OnThink!